python/ray/train/utils.py

```diff
@@ -33,33 +33,35 @@
 logger = logging.getLogger(__name__)
 
 
-def check_for_failure(remote_values: List[ObjectRef]) -> Tuple[bool, List[int]]:
+def check_for_failure(remote_values: List[ObjectRef]) -> bool:
     """Check for actor failure when retrieving the remote values.
 
     Args:
         remote_values (list): List of object references from Ray actor methods.
 
     Returns:
-        Returns Tuple of success boolean and list of workers indexes that fail.
+        True if evaluating all object references is successful, False otherwise.
     """
     unfinished = remote_values.copy()
-    dead_worker_indexes = []  # Store the indexes of the failed workers.
-    while len(unfinished) > 0:
+
+    at_least_one_failed_worker = False
+    while len(unfinished) > 0 and not at_least_one_failed_worker:
         finished, unfinished = ray.wait(unfinished)
+
         # If a failure occurs the ObjectRef will be marked as finished.
         # Calling ray.get will expose the failure as a RayActorError.
         for object_ref in finished:
+            # Everything in finished has either failed or completed
+            # successfully.
             try:
                 ray.get(object_ref)
             except RayActorError as exc:
                 logger.exception(str(exc))
                 failed_actor_rank = remote_values.index(object_ref)
                 logger.info(f"Worker {failed_actor_rank} has failed.")
-                dead_worker_indexes.append(failed_actor_rank)
-    if len(dead_worker_indexes) > 0:
-        return False, dead_worker_indexes
-    else:
-        return True, []
+                at_least_one_failed_worker = True
```

Comment:
nit: Just return `False` here? Or remove `and not at_least_one_failed_worker` in line 48 if you want to print all failed workers.

Reply:
good point, updated!